client/src/controllers/MessagesController.ts

```diff
@@ -1,29 +1,17 @@
+
 import { Controller } from '@hotwired/stimulus';
 
-/**
- * Adds the ability for a controlled element to pick an element from a template
- * and then clone that element, adding it to the container.
- * Additionally, it will allow for clearing all previously added elements.
- *
- * @example
- * <div
- *   data-controller="w-messages"
- *   data-action="w-messages:add@document->w-messages#add"
- *   data-w-messages-added-class="new"
- *   data-w-messages-show-class="appear"
- * >
- *   <ul data-w-messages-target="container"></ul>
- *   <template data-w-messages-target="template">
- *     <li data-message-status="error-or-success"><span></span></li>
- *  </template>
- * </div>
- */
-export class MessagesController extends Controller<HTMLElement> {
-  static classes = ['added', 'show'];
-  static targets = ['container', 'template'];
-  static values = {
-    showDelay: { default: 100, type: Number },
-  };
+declare global {
+  interface Window {
+    handleUnsavedEvent: (event: CustomEvent<{
+      formDirty: boolean;
+      commentsDirty: boolean;
+    }>) => void;
+  }
+}
+export default class extends Controller {
+  static targets = ["container", "template"];
+  static values = { clearDelay: Number ,clearDelayValue: Number};
```

Comment:
for some reason you have removed the `showDelay`, also it would be good to ensure that all values have a default, otherwise the existing HTML usage in the page level messages using this will throw an error.

See https://stimulus.hotwired.dev/reference/values#default-values